galsim/sed.py

```diff
@@ -818,6 +818,49 @@ def calculateSeeingMomentRatio(self, bandpass, alpha=-0.2, base_wavelength=500):
             return galsim.integ.int1d(lambda w: weight(w) * kernel(w),
                                       bandpass.blue_limit, bandpass.red_limit) / flux
 
+    def sampleWavelength(self, nphotons, bandpass, seed=0, npoints=256):
+        """ Sample 'nphoton' random wavelengths from the SED (and bandpass).
```

Comment:
Minor point, but the quoted 'nphoton' here is not the actual name of the parameter (nphotons), but I think it might read better to just use plain text here.  Something like "Sample a number of random wavelength values from the SED, possibly as observed through a bandpass."